src/json.cpp

```diff
@@ -1020,32 +1020,84 @@ std::string JsonIn::get_string()
     throw JsonError( "something went wrong D:" );
 }
 
+number_sci_notation JsonIn::get_any_int()
+{
+    number_sci_notation n = get_any_number();
+    if( n.exp < 0 ) {
+        error( "Integers cannot have a decimal point or negative order of magnitude." );
+    }
+    // Manually apply scientific notation, since std::pow converts to double under the hood.
+    for( int i = 0; i < n.exp; i++ ) {
+        if( n.number > UINT64_MAX / 10ULL ) {
+            error( "Specified order of magnitude too large -- encountered overflow applying it." );
+        }
+        n.number *= 10ULL;
+    }
+    n.exp = 0;
+    return n;
+}
+
 int JsonIn::get_int()
 {
-    // get float value and then convert to int,
-    // because "1.359e3" is technically a valid integer.
-    return static_cast<int>( get_float() );
+    number_sci_notation n = get_any_int();
+    if( !n.negative && n.number > INT_MAX ) {
+        error( "Found a number greater than " + std::to_string( INT_MAX ) +
+               " which is unsupported in this context." );
+    } else if( n.negative && n.number > ( UINT_MAX - INT_MAX ) ) {
+        error( "Found a number less than " + std::to_string( INT_MIN ) +
```

Comment:
This reports `INT_MIN` in the error message, but the test is against a different term. If it's guaranteed that both evaluate to the same number, why use different terms in the code? If the numbers might be different (`UINT_MAX - INT_MAX` is not guaranteed to be the same as `INT_MIN`), the error will report a different error condition than what is checked.

Reply:
`UINT_MAX - INT_MAX` is guaranteed to evaluate to the absolute value of `INT_MIN`. I can't do `-INT_MIN` because of overflow, and I can't do `-( UINT_MAX - INT_MAX )` because an `unsigned int` can't be negative.

The only option I see to reconcile your issue with the type limitations, is to reframe the message like this:
```cpp
error( "Found a number whose absolute value is more than " +
       std::to_string( UINT_MAX - INT_MAX ) +
       " which is unsupported in this context." )
```
Which I think is too convoluted to be reasonable.

Reply:
Never mind, Kevin enlightened me to `numeric_limits<int>::min()`. Will fix this shortly.

Reply:
Err, nevermind again, I don't know the solution to this, and now I'm not even sure that this:
> `UINT_MAX - INT_MAX` is guaranteed to evaluate to the absolute value of `INT_MIN`.

Is accurate. Still looking for a solution.

Reply:
Okay, I think I've solved this, would appreciate a second look if you have time.